### ISSUE TITLE
inconsistent behavior in TB115.2.2

### ISSUE THREAD

**Issue by Elliott599** [cleidigh/EditEmailSubject-MX#81]
I goofed.  I didn't mean to update to 115 so soon.  I was going to wait 2 months.  But here I am in 115.  In 102 your addon worked fine.  Now, not so much.

Sometimes it works just fine.  The behavior is different than in 102, in that a small screen pops up with a rotating arrow.  At the end of that sometimes I get a green check mark and all is good.  The edited message has replaced the original.

But 2/3 of the time at the end of that I get a red x.  When that happens I can press get mail and then the changed version downloads but I still have the original version in my mailbox.

I have a compuserve imap account and also comcast imap accounts.  It works with the same intermittency in both accounts.  I have closed and reopened the program a few times and it still behaves the same.

I use your program all the time.  Please help.

**Comment by jobisoft** [comment]
This has nothing to do with 115. I simply changed the method of how I work with messages in 115. Looks like your server does not like it. 

The issue is, that for whatever reason, your inbox does not show the renamed message and because the add-on does not see the renamed version, it does not finish the process and deletes does not the original message.

**Comment by Elliott599** [comment]
It is not just that it does not delete the original message, but also it does not show the edited message until I hit "get mail."  Then both messages are present.  Also, the process takes many seconds; it used to be very quick.

I do not believe it is a server problem since I have 3 accounts each using a different server.  My compuserserve account uses imap.aol.com, my comcast account uses imap.comcast.com and my gmail account uses imap.gmail.com.  I experience the problem on all 3 accounts.  I also experience it on 3 different computers.  Two of them are running the latest version of win10 and one is running win7.

I have used your program for a few years now in prior versions of TBird and it was flawless.  Please take another look and see what you can do to fix this issue.

At the end of your note you wrote about the error console.  Please let me know how to find that if indeed you are asking for me to send you info from that console.

Thank you.

**Comment by jobisoft** [comment]
> It is not just that it does not delete the original message, but also it does not show the edited message ..."

That is why I wrote: The issue is, that for whatever reason, your inbox does not show the renamed message.
 
> I do not believe it is a server problem since I have 3 accounts each using a different server. My compuserserve account uses imap.aol.com, my comcast account uses imap.comcast.com and my gmail account uses imap.gmail.com. I experience the problem on all 3 accounts. I also experience it on 3 different computers. Two of them are running the latest version of win10 and one is running win7.

It could be that you have a special setting, which disabled auto-sync with your IMAP servers. I have no details at this point, just guessing.

> At the end of your note you wrote about the error console. Please let me know how to find that if indeed you are asking for me to send you info from that console.

The error console should open, when you press CTRL+SHIFT+J when the main Thunderbird window is active.

**Comment by jobisoft** [comment]
Hm, can you open the Add-on Manager, select "Debug add-ons" from the gear menu and click on the "Inspect" button in the Quicktext entry? That should also open a console. Any interesting entries there?

**Comment by Elliott599** [comment]
The inspect button gives me a console with many items.  Not sure what you want.  I don't see how to attach images in github so I will reply to your email with image captures.

[I](url) get 3 options, extension ID, Internan UUID, and Manifest URL.  Here is what they say:
Extension ID - EditMailSubject@jcp.convenat
Internal UUID - 04d85540-f80c-4aa2-8ad4-4724fe382d44
Manifect URL leads me to this window pasted below.

{
  "applications": {
    "gecko": {
      "id": "EditMailSubject@jcp.convenant",
      "strict_min_version": "115.0"
    }
  },
  "manifest_version": 2,
  "name": "EditEmailSubject MX",
  "version": "7.4",
  "author": "John Bieling",
  "description": "__MSG_extensionDescription__",
  "default_locale": "en-US",
  "icons": {
    "64": "content/images/editemailsubjectmx-icon-64px.png",
    "32": "content/images/editemailsubjectmx-icon-32px.png",
    "16": "content/images/editemailsubjectmx-icon-16px.png"
  },
  "permissions": [
    "storage",
    "menus",
    "messagesRead",
    "messagesMove",
    "messagesImport",
    "accountsRead",
    "accountsFolders",
    "messagesDelete"
  ],
  "background": {
    "page": "background.html"
  },
  "commands": {
    "edit_email_subject": {
      "suggested_key": {
        "default": "Ctrl+Shift+E"
      },
      "description": "Open the Edit Email Subject popup"
    }
  },
  "options_ui": {
    "page": "/content/options/options.html",
    "browser_style": true
  }
}

**Comment by Elliott599** [comment]
If you have any other questions I will be off line till about 3 pm EDT.  Thank you for your quick efforts on this.

**Comment by jobisoft** [comment]
It could look like so:

![image](https://github.com/cleidigh/EditEmailSubject-MX/assets/5830621/7599730c-94ad-4fe0-b932-0bf129ebacbe)

I think I encountered the same issue while trying to generate the screenshot.

**Comment by Elliott599** [comment]
Not sure what you mean by "encountered the same issue."  Do you mean that the change subject failed?

I went back to inspect and the Console was empty.  I turned on the debugger and tried a few changes.  All of the ones in my comcast inbox succeeded and the ones I did in my compuserve/AOL in box failed.  Here is what Console showed?  What else can I do for you.  


![Console](https://github.com/cleidigh/EditEmailSubject-MX/assets/76265808/62a2f484-7cf1-4883-8596-e21b6c5cd0d6)

If this image of my console does not come thru let me know and I will email to you.

**Comment by jobisoft** [comment]
> Not sure what you mean by "encountered the same issue." Do you mean that the change subject failed?

Yes. I wanted to confirm, that I was able to reproduce the issue you are describing with my google account.

A small difference is, that I do not have to "press get mail", but instead I can wait a few more seconds and the message will appear eventually. I currently have no answer as to why the message has such a long delay, before it appears in the inbox.

I do not need more information from you. Thanks for your feedback. I have to think about this.

**Comment by Elliott599** [comment]
Thanx much.  Let me know if there is any more info I can provide.  I eagerly await your solution.

**Comment by TrikeBike** [comment]
Hello, reading with interest..   2 identical computers one works the other doesnt ?

Just stays showing the rotating circle for hours if you let it. ?

I have looked at the error log as you have explained above and it shows the following message

NotFoundError: No such JSWindowActor 'DevToolsFrame'

Any ideas ? Of course it is the computer that needs to change the email subject is the one where it is not working properly. Thank you for your help

PS have tried it with the "backup email into the local folder" selected and unselected

**Comment by jobisoft** [comment]
> Just stays showing the rotating circle for hours if you let it. ?

I have implemented a timeout, so it will abort and show you a red x after some time. I am currently playing with extending the timeout (which makes it work for me again). But I am also trying to understand, how I can force the folder to update. I am currently waiting for "something" to update the folder after I modified the message and I would like to trigger that manually.

**Comment by exapplerep** [comment]
I believe I'm seeing essentially the same issue here. When I "OK" the dialog, I see the spinner and then eventually the "X". The original message remains in the inbox. I dismiss the dialog. I press "Get Mail" and the changed eMail appears in the list (_and the original is still there_ although it obviously should have been deleted). Same result with both my Google and Yahoo eMail.

I suspect the changed eMail would appear if I simply waited the 5 minutes for the next time TBird checked for mail.

BTW: If I attempt to change the subject of the _changed_ eMail (the one that now has the new subject), the dialog will show the original subject as the text of the "real" ORIGINAL eMail. Let me provide an example:
- Original eMail subject: "My Received eMail"
- I invoke the addon and change the subject to "My Renamed eMail" (and the dialog shows the original subject which is expected).
- I invoke the addon again and the original subject is shown as "My Received eMail" (even though it ws changed to "My Renamed eMail").

I suspect the above is the proper behavior as I think I remember an earlier thread about how eMails have all sorts of metadata hidden from the eyes of normal users and that "original" subject may be persistent regardless of how your addon changes what _we_ see.

Anyway, I will be patient as your addon is the only one that did the job perfectly (and still mostly works even if I have to help it along manually). Thanks very much!

Barry

**Comment by WoofGrrrr** [comment]
I am using EES MX 7.4 on TBird 115.8.1

I am seeing the same issues.  I click the "OK" button on the ESS dialog, the busy icon spins for about 10 seconds, the red X shows very briefly, then the dialog remains.

I click the "Get Messages" button and the message with the new subject finally appears, but the original is still there as well.  There is nothing in the ESS-Temp Local Folder.  The "Back up email into the local folder..." checkbox IS checked.

**I copied the following from the Console in the  inspector:**

```
Created [8 -> 9] editemailsubject.mjs:136:11
Error: Message not found after rename
    checkFolder moz-extension://fa3d56dc-0515-4932-8280-bf6627d275a7/content/scripts/editemailsubject.mjs:162
editemailsubject.mjs:171:13
    updateMessage moz-extension://fa3d56dc-0515-4932-8280-bf6627d275a7/content/scripts/editemailsubject.mjs:171
```

**Comment by jobisoft** [comment]
I prepared some fixes to the IMAP handling code of the messages API. Once they landed, I can merge [this](https://github.com/cleidigh/EditEmailSubject-MX/commit/e4503d4f1a37bca6f2d37e48d66c30102f151aab) and release a new version of EES. That should fix this issue as well.
